flocker/node/agents/ebs.py

```diff
@@ -1022,20 +1016,30 @@ def _next_device(self, instance_id, volumes, devices_in_use):
         :returns ``None`` if suitable EBS device names on this EC2
             instance are currently occupied.
         """
-        volume_devices = []
-        for v in volumes:
-            volume_attachments = v.attachments
-            for attachment in volume_attachments:
-                if attachment['InstanceId'] == instance_id:
-                    volume_devices.append(attachment['Device'])
-        devices = pset(volume_devices)
-        devices = devices | devices_in_use
+
+        command = [b"/bin/lsblk", b"--output", b"KNAME"]
+        command_result = check_output(command)
+        local_devices = pset(filter(
+            lambda d: d.startswith(b"xvd") or d.startswith('sd'),
+            command_result.split("\n")[1:]
+        ))
+        devices_in_use = pset(list(devices_in_use) + list(
+            device.replace('/dev/sd', 'xvd') for device in devices_in_use
+        ))
+        devices = local_devices | devices_in_use
         sorted_devices = sorted(list(thaw(devices)))
         IN_USE_DEVICES(devices=sorted_devices).write()
 
         for suffix in b"fghijklmonp":
+            next_local_device = b'xvd' + suffix
+            next_local_sd_device = b'sd' + suffix
             file_name = u'/dev/sd' + suffix
-            if file_name not in devices:
+            possible_devices = [
+                next_local_device, next_local_sd_device, file_name
```

Comment:
We really shouldn't have `possible_devices` contain all three of `/dev/sda`, `xvda` and `sda`. We should pick one and stick to it.